plugin/cache.go

```diff
@@ -144,7 +149,13 @@ func (c DuplicateCNameErr) Error() string {
 // querying a specific entry.
 func CachedList(ctx context.Context, g Group) (map[string]Entry, error) {
```

Comment:
Would also be useful to ad a comment [somewhere here](https://github.com/puppetlabs/wash/blob/master/plugin/cache.go#L169) that we're setting these IDs to ensure that all entries have an ID, even those that didn't need any of the Cached* methods in their parent's List call.